### PR TITLE
FT2-293 version comparison base module and ddls

### PR DESCRIPTION
**Pull request by andycyberian**
Added new version comparison module
Create new compare version admin page
Added version compare drop down list to select versions
Update recipes
Added Compare version link to manage content items page

### CODE REVIEW

**Review by ravi-methods — APPROVED** [review]

File: DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            _auditTrailQueryService = auditTrailQueryService;
+            _contentManager = contentManager;
+            _h = htmlLocalizer;
+            _notifier = notifier;
+            _shapeFactory = shapeFactory;
+            _updateModelAccessor = updateModelAccessor;
+            _versionComparisonOptionsDisplayManager = versionComparisonOptionsDisplayManager;
+        }
+
+        public async Task<ActionResult> Index(string contentItemId)
```
Comment (ravi-methods) [inline DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs:46]: Should stick a [HttpGet] here?

File: DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                });
+        }
+
+        private List<SelectListItem> BuildSelectList(List<int> versionNumbers)
+        {
+            var selectListItems = new List<SelectListItem>();
+            for (int i = 0; i < versionNumbers.Count; i++)
+            {
+                var versionNumberValue = versionNumbers[i].ToString();
+                var versionNumberText = string.Format("version {0}{1}", versionNumberValue,
```
Comment (ravi-methods) [inline DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs:120]: string interpolation?

File: DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+using Microsoft.AspNetCore.Mvc.Localization;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using OrchardCore.ContentManagement;
+using OrchardCore.DisplayManagement;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.DisplayManagement.Notify;
+
+namespace DFC.ServiceTaxonomy.VersionComparison.Controllers
+{
+    public class AdminController : Controller
```
Comment (pramireddy) [inline DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs:17]: Do we need here  Authorize:Admin attribute?

File: DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs (hunk 1 above)
Comment (andycyberian) [reply DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs:47]: Not usually done in OC core code as is implied as a GET by default but have added in our module so is explicit.

File: DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs (hunk 2 above)
Comment (andycyberian) [reply DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs]: A bit subjective which one is more readable with the ternery operation but have changed to string interpolation.

File: DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs (hunk 3 above)
Comment (andycyberian) [reply DFC.ServiceTaxonomy.VersionComparison/Controllers/AdminController.cs:17]: I don't believe so.

**Pull request merged**